components/sensor/index.rst

```diff
@@ -37,7 +37,11 @@ override them if you want to.
 
 Configuration variables:
 
-- **name** (**Required**, string): The name for the sensor.
+- **name** (**Required**, string): The name for the sensor.  Caution in setting this name as it forms 
```

Comment:
This addition looks incorrect; or, maybe I'm misunderstanding the reasoning for this change:

Example:

```yaml
sensor:
  - platform: uptime
    name: "OMG & PUPPIES 2.5, high %!!"
```

Produces:

![image](https://user-images.githubusercontent.com/195327/117565046-a6416100-b0af-11eb-85f0-0a34fe6715bd.png)

Reply:
Hi Franck,

Thanks for taking time to look into this.

Attached to this email is the YAML code that  has the clashing lines:
```
    sensors:
    - name: "PMS5003 Dust Sensor PM1.0"      <<<<<<<<<<<<<<<<<<
      unit_of_measurement: "ug/m3"
      accuracy_decimals: 0 
    - name: "PMS5003 Dust Sensor PM2.5"
      unit_of_measurement: "ug/m3"
      accuracy_decimals: 0
    - name: "PMS5003 Dust Sensor PM10"       <<<<<<<<<<<<<<<<<<
      unit_of_measurement: "ug/m3"
```

It isn't so much that the entities in Home Assistance do not show "." (see the following) Screenshot HA Entities.png

It is that it looks like ESPHome creates some unique ID for each entity and constructs this from the name per my push request.
If I use the above YAML configuration I get the following Log Error message indicating the name clash.
```
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:541
Integration: Sensor (documentation, issues)
First occurred: 21:05:48 (1 occurrences)
Last logged: 21:05:48

Platform esphome does not generate unique IDs. ID shiny-inventionsensorpms5003_dust_sensor_pm10 already exists - ignoring sensor.pms5003_dust_sensor_pm10
```


If I use the following it works OK:
```
    sensors:
    - name: "PMS5003 Dust Sensor PM1"       <<<<< This is changed
      unit_of_measurement: "ug/m3"
      accuracy_decimals: 0 
    - name: "PMS5003 Dust Sensor PM2.5"
      unit_of_measurement: "ug/m3"
      accuracy_decimals: 0
    - name: "PMS5003 Dust Sensor PM10"
      unit_of_measurement: "ug/m3"
      accuracy_decimals: 0
```

So my thinking is per my request - just put some guidance in the docs rather than try and do any special error checking; this is pretty much a corner-case that I was unlucky enough to hit.

BTW - this has been fun using the ESPHome project for some environmental monitoring and control stuff.

The code attached is the customised TMS5003 that I've done for completeness; but I don't expect it to be at issue with the feature I've encountered.

Cheers
Tony Saul
Brisbane, Australia.


![Screenshot HA Entities](https://user-images.githubusercontent.com/36117102/117570399-4083c200-b10d-11eb-9c72-353133499e99.png)
![Screenshot- HA Log Error Message](https://user-images.githubusercontent.com/36117102/117570400-41b4ef00-b10d-11eb-8d58-c4036ff43811.png)
[shiny-invention.zip](https://github.com/esphome/esphome-docs/files/6447369/shiny-invention.zip)
[airQualityPMS5003.zip](https://github.com/esphome/esphome-docs/files/6447370/airQualityPMS5003.zip)